build-logic/jvm/src/main/kotlin/gradlebuild.launchable-jar.gradle.kts

```diff
@@ -14,59 +14,33 @@
  * limitations under the License.
  */
 
-import gradlebuild.startscript.tasks.GradleStartScriptGenerator
+import gradlebuild.configureAsJarClasspath
 
 plugins {
     java
 }
 
-val manifestClasspath by configurations.creating {
-    isTransitive = false
-
-    configureAsJarClasspath()
+interface LaunchableJar {
+    /**
+     * The main class for the application. Can be undefined.
+     */
+    val mainClassName: Property<String>
```

Comment:
This plugin is now used by multiple projects, so the main class name needs to be configurable.